src/linearcombination.jl

```diff
@@ -60,6 +62,67 @@ function A_mul_B!(y::AbstractVector, A::LinearCombination, x::AbstractVector)
     return y
 end
 
+else # 5-arg mul! is available for matrices
+
+# map types that have an allocation-free 5-arg mul! implementation
+const FreeMap = Union{MatrixMap,UniformScalingMap}
+
+function A_mul_B!(y::AbstractVector, A::LinearCombination, x::AbstractVector)
+    # no size checking, will be done by individual maps
+    A_mul_B!(y, A.maps[1], x)
+    l = length(A.maps)
+    if l>1
+        z = similar(y)
+        for n in 2:l
+            An = A.maps[n]
+            if An isa FreeMap
+                mul!(y, An, x, true, true)
+            else
+                A_mul_B!(z, A.maps[n], x)
+                y .+= z
+            end
+        end
+    end
+    return y
+end
+function A_mul_B!(y::AbstractVector, A::LinearCombination{T,As}, x::AbstractVector) where {T, As<:Tuple{Vararg{FreeMap}}}
```

Comment:
I did not see this yet, but it would then become unnecessary with the trick above.

Reply:
On the other hand, the current implementation is also very Julian, so just keep it like it is. Maybe just change the order of these two methods, such that one first reads the implementation of the more specific method and then the one of the more general fallback method.